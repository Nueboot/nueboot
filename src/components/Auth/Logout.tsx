import React from 'react';
import styled from 'styled-components';

import { Dropdown, DropdownContent } from '../Styled/Dropdown';
import Text from '../Styled/Text';

interface LogoutProps {
  readonly displayName: string | null;
}

const HoverText = styled(Text)`
  :hover {
    font-weight: 700;
  }
`;

const Logout: React.SFC<LogoutProps> = ({ displayName }) => (
  <Dropdown>
    <Text>{displayName} ‚åµ</Text>
    <DropdownContent>
      <HoverText>Sign out</HoverText>
    </DropdownContent>
  </Dropdown>
);

export default Logout;
