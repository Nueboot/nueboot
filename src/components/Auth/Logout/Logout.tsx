import React from 'react';
import styled from 'styled-components';

import { Text } from '../../Styled';
import { Dropdown, DropdownContent } from '../../Styled/Dropdown';

interface LogoutProps {
  readonly displayName: string;
  logout(): void;
}

const HoverText = styled(Text)`
  :hover {
    font-weight: 700;
  }
`;

const Logout: React.SFC<LogoutProps> = ({ displayName, logout }) => (
  <Dropdown>
    <Text tabIndex={0}>{displayName} ‚åµ</Text>
    <DropdownContent>
      <HoverText tabIndex={0} onClick={logout}>
        Log out
      </HoverText>
    </DropdownContent>
  </Dropdown>
);
export default Logout;
