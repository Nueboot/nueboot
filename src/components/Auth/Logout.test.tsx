import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import Logout from './Logout';

describe('Logout', () => {
  let logoutComponent: ShallowWrapper;

  beforeEach(() => {
    logoutComponent = shallow(<Logout displayName="Mickey" />);
  });

  it('renders correctly', () => {
    expect(logoutComponent).toMatchSnapshot();
  });
});
