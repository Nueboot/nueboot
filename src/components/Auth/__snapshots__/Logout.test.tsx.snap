// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout renders correctly 1`] = `
<styled.div>
  <styled.span>
    Mickey
     ‚åµ
  </styled.span>
  <styled.div>
    <Styled(styled.span)>
      Sign out
    </Styled(styled.span)>
  </styled.div>
</styled.div>
`;
